### PR TITLE
Add tutorial for setting up ROS on A1

### PR DESCRIPTION
**Pull request by yxyang**
^title

### CODE REVIEW

File: Tutorials/a1_ros_setup.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+nav_order: 2 
+mathjax: true
+---
+
+**Last Updated: 2021-10-04**
+
+# Setting up ROS for A1
+This tutorial goes through steps to set up the [Robot Operating System (ROS)](https://www.ros.org/) pipeline for the A1 quadruped robot. 
+In our case, we use a Mac-Mini with M1 processor (running Ubuntu 20.04 Virtual Machine using the Parallels) for high-frequency model predictive control,
+and the on-board Nvidia Jetson TX2 (running Ubuntu 18.04) for perception and other tasks.
```
Comment (romesco) [inline Tutorials/a1_ros_setup.md:14]: Keeping the TX2 for perception or will this be updated eventually?

File: Tutorials/a1_ros_setup.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+While ROS is not necessarily required for the operation of A1, we choose ROS for a number of benefits:
+
+1. Distributed, parallel computation with easy inter-process message passing.
+
+Note that different components of the A1's control pipeline are inherently asynchronous (e.g. camera perception process might run at a significantly lower frequency compared to the control process) and distributed (e.g. we run perception on TX2 to utilize its GPU, and run control on M1 to utilize its CPU). Moreover, components of the pipeline should be interchangable (e.g. one should easily replace an automatic SLAM-based controller with a manual controller). This calls for easy inter-process and inter-machine message passing, which is one of the core features of ROS.
+
+2. Automated logging for easy debugging.
+
+Manually implements the logging of every component in the control pipeline can be tedious. Instead, ROS can automatically log all topics via [rosbag](http://wiki.ros.org/rosbag), and most of the screen logs are also automatically saved, which significantly simplifies debugging.
+Another
```
Comment (romesco) [inline Tutorials/a1_ros_setup.md]: Another ???

File: Tutorials/a1_ros_setup.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+2. Automated logging for easy debugging.
+
+Manually implements the logging of every component in the control pipeline can be tedious. Instead, ROS can automatically log all topics via [rosbag](http://wiki.ros.org/rosbag), and most of the screen logs are also automatically saved, which significantly simplifies debugging.
+Another 
+
+3. Easy integration with mobile devices for outdoor testing.
+
+It can be challenging to carry a laptop when testing the robot outdoors. Instead, it would be much easier if one could view the robot's state on a phone. Thankfully, the [ROS-Mobile](https://github.com/ROS-Mobile/ROS-Mobile-Android) project allows you to do exactly this on Android. While it doesn't support all message types yet, it's already supporting image and text, and works nicely with ROS-Noetic.
+
+Also, keep in mind that, despite its benefits, ROS1 is **not** feasible for high-frequency, real-time computation. The core message passing is based on [xmlrpc](http://xmlrpc.com/), which is optimized more for compatiability than for performance. Therefore, I do not use ROS for low-level model predictive control.
```
Comment (romesco) [inline Tutorials/a1_ros_setup.md:32]: This is because ROS1 lacks real-time guarantees, correct?

File: Tutorials/a1_ros_setup.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+## Make ROS processes run at boot time
+Since the robot can suffer from unexpected conditions and stop/reboot randomly esp. during outdoor operation, it can be painful to re-connect to the robot every time and manually start the ROS processes. There are several ways to make scripts run at start-up in Ubuntu, including [`init.d`](https://en.wikipedia.org/wiki/Init), [`systemd`](https://en.wikipedia.org/wiki/Systemd), [`cron` @reboot](https://phoenixnap.com/kb/crontab-reboot) and [`autostart`](https://arcolinux.com/how-to-autostart-any-application-on-any-linux-desktop/). They operate at different levels. At the abstract level, `init.d` runs at OS bootup. `systemd` runs as a service when OS is running. `cron` is a `systemd` service that allows more flexible scheduling. `autostart` is Ubuntu-specific that's tied to Gnome.
+
+Since ROS requires networking, putting it in `init.d` requires careful ordering and can be tricky. `autostart` and `cron` don't have good support on defining custom environment variables or sourcing other bash scripts, which is required for ROS. Therefore, I have found `systemd` to be the easiest way to configure this.
```
Comment (romesco) [inline Tutorials/a1_ros_setup.md:72]: This is very helpful info... I've always run into arbitrary logic for which to choose.

File: Tutorials/a1_ros_setup.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+[Unit]
+Description=YOUR DESCRIPTION HERE
+
+[Service]
+ExecStart=ABSOLUTE PATH TO YOUR BASH SCRIPT
+User=YOUR USER NAME, OR LEAVE EMPTY AND IT WILL BE RUN UNDER ROOT
+
+[Install]
+WantedBy=multi-user.target
+```
```
Comment (romesco) [inline Tutorials/a1_ros_setup.md:85]: Perhaps give the example for the services you are starting for the A1 in a block following this 'example block'?

**Review by romesco — APPROVED** [review]
Very useful and pretty much ready to go as is. Just a few clarification questions / non urgent suggestions.

File: Tutorials/a1_ros_setup.md (hunk 1 above)
Comment (yxyang) [reply Tutorials/a1_ros_setup.md:14]: Will likely have to be TX2 for a while, unless Unitree is able to upgrade their built-in hardware.

File: Tutorials/a1_ros_setup.md (hunk 2 above)
Comment (yxyang) [reply Tutorials/a1_ros_setup.md]: Fixed, thanks.

File: Tutorials/a1_ros_setup.md (hunk 3 above)
Comment (yxyang) [reply Tutorials/a1_ros_setup.md:32]: Yes. It is in fact very bad in terms of real-time guarantee.

File: Tutorials/a1_ros_setup.md (hunk 5 above)
Comment (yxyang) [reply Tutorials/a1_ros_setup.md:85]: The example I'm using for A1 is changing very rapidly these days. I'll probably put them in the documentation in my code repo.

**Pull request merged**